### PR TITLE
Require types in tests.storage.

### PR DESCRIPTION
**Pull request by clokep**
I didn't break up these commits very well, but it enforces types in `tests.storage`.

### CODE REVIEW

**Comment by DMRobertson** [comment]
Oh, I can hold off looking at this if you want to break it up into commits etc. (But I also don't mind going through it in a big blob)

**Comment by clokep** [comment]
> Oh, I can hold off looking at this if you want to break it up into commits etc. (But I also don't mind going through it in a big blob)

I suspect just going file by file and marking them as done will work fine for this PR. trial-olddeps failed, so this likely needs a small tweak, but I think can still be reviewed.

File: tests/storage/test_appservice.py [hunk 1]
```diff
@@ -149,15 +151,17 @@ def _add_service(self, url, as_token, id) -> None:
...
             self.engine.convert_param_style(
                 "INSERT INTO application_services_state(as_id, state) VALUES(?,?)"
             ),
             (id, state.value),
         )
 
-    def _insert_txn(self, as_id, txn_id, events):
+    def _insert_txn(
+        self, as_id: str, txn_id: int, events: List[Mock]
+    ) -> defer.Deferred:
```
Comment (DMRobertson) [inline tests/storage/test_appservice.py:164]: Possibly use `Deferred[None]` here? (Per https://docs.twistedmatrix.com/en/stable/api/twisted.enterprise.adbapi.ConnectionPool.html#runOperation )

File: tests/storage/test_cleanup_extrems.py [hunk 2]
```diff
@@ -286,7 +295,7 @@ def test_send_dummy_event(self):
         self.assertTrue(len(latest_event_ids) < 10, len(latest_event_ids))
 
     @patch("synapse.handlers.message._DUMMY_EVENT_ROOM_EXCLUSION_EXPIRY", new=0)
```
Comment (DMRobertson) [inline tests/storage/test_cleanup_extrems.py:297]: huh, didn't know you can `@patch` an entire test method!

File: tests/storage/test_redaction.py [hunk 3]
```diff
@@ -11,27 +11,35 @@
...
         config["redaction_retention_period"] = "30d"
         return config
 
-    def prepare(self, reactor, clock, hs):
+    def prepare(self, reactor: MemoryReactor, clock: Clock, hs: HomeServer) -> None:
         self.store = hs.get_datastores().main
-        self._storage = hs.get_storage_controllers()
+        storage = hs.get_storage_controllers()
+        assert storage.persistence is not None
+        self._persistance = storage.persistence
```
Comment (DMRobertson) [inline tests/storage/test_redaction.py:42]: Possible `self._persistence` with an `e`? Not that it makes a huge difference...

**Review by DMRobertson — APPROVED** [review]
Thanks Patrick. Looks like this was fairly routine and there weren't any horrible surprises for you to grapple with!

A few minor nits, happy for you to leave them both as-is though.

File: tests/storage/test_redaction.py (hunk 3 above)
Comment (clokep) [reply tests/storage/test_redaction.py:42]: Oops!

**Pull request merged**